File: 2887-sort-vowels-in-a-string/2887-sort-vowels-in-a-string.cpp
class Solution {
public:
    string sortVowels(string s) {
        map<char,bool>m;
        vector<char>v = {'A','E', 'I','O','U', 'a', 'e', 'i', 'o', 'u' };
        for(auto i:v)
        {
            m[i]=true;
        }
        int n = s.size();
        vector<char>vec;
        for(int i=0;i<n;i++)
        {
            if(m[s[i]])
            {
                vec.push_back(s[i]);
                s[i]='#';
            }
        }
        sort(vec.begin(), vec.end());
        int pos=0;
        for(int i=0;i<n && pos<vec.size();i++)
        {
            if(s[i]=='#')
            {
                s[i]=vec[pos++];
            }
        }
        return s;
    }
};

File: 2887-sort-vowels-in-a-string/2887-sort-vowels-in-a-string.java
class Solution {
    public static boolean isVowel(Character c)
    {
        return c == 'a' || c == 'e' || c == 'o'|| c == 'u'|| c == 'i'
                || c == 'A' || c == 'E' || c == 'O'|| c == 'U'|| c == 'I';
    }
    public String sortVowels(String s) {
        int[] count = new int[200];
        int n = s.length();
        for(char c : s.toCharArray())
        {
            if(isVowel(c))
            {
                count[c-'A']++;
            }
        }
        String vwls = "AEIOUaeiou";
        StringBuilder ans = new StringBuilder();
        int j=0;
        for(int i=0;i<n;i++)
        {
            if(!isVowel(s.charAt(i)))
            {
                ans.append(s.charAt(i));
            }
            else
            {
                while(count[vwls.charAt(j)-'A'] ==0)
                {
                    j++;
                }
                ans.append(vwls.charAt(j));
                count[vwls.charAt(j)-'A']--;
            }
        }
        return ans.toString();
    }
}